File: Sandbox_Java/src/main/java/spark/utils/SparkUtils.java
package spark.utils;

import org.apache.log4j.Level;
import org.apache.log4j.Logger;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.streaming.Duration;
import org.apache.spark.streaming.api.java.JavaStreamingContext;

public class SparkUtils implements Constantes{
	
	public static JavaSparkContext initContext(String appName){
		// Dependency needed because of Windows
		System.setProperty("hadoop.home.dir", WINUTIL_PATH);

		// Cut some Spark logs
		Logger.getLogger("org").setLevel(Level.OFF);
		Logger.getLogger("akka").setLevel(Level.OFF);

		// Initialize Spark Streaming context
		SparkConf conf = new SparkConf().setAppName(appName).setMaster("local[*]");
		return new JavaSparkContext(conf);
	}
	
	public static JavaStreamingContext initStreamingContext(String appName, Duration bachDuration){
		return new JavaStreamingContext(initContext(appName), bachDuration);
	}

}

File: Sandbox_Scala/src/spark/SparkUtils.scala
package spark

import org.apache.log4j.{ Level, Logger }
import org.apache.spark.streaming.{ Seconds, StreamingContext }
import org.apache.spark._

object SparkUtils {

        def initSpark(): SparkContext = {

                Logger.getLogger("org").setLevel(Level.OFF);
                Logger.getLogger("akka").setLevel(Level.OFF);
                val conf: SparkConf = new SparkConf()
                        .setAppName("TestGraphX")
                        .setMaster("local[*]")

                System.setProperty("hadoop.home.dir", "c:\\winutil\\") // TODO need an hadoop dependency for checkpoint/spark logs : bug ?
                return new SparkContext(conf)
        }

        def initSparkStreaming(sc: SparkContext): StreamingContext = {
                new StreamingContext(sc, Seconds(5))
        }

        def closeSpark(sc: SparkContext) {
                sc.stop()
        }

        def startStreaming(ssc: StreamingContext) {
                ssc.start()
                ssc.awaitTermination()
        }
}